### DISCUSSION TITLE
Could not resolve all files for configuration ':capacitor-community-bluetooth-le:classpath'

### DISCUSSION THREAD

**Discussion by mayte130** [capacitor-community/bluetooth-le#561] [Q&A] [answered]
Hello!
I have an issue when running the command "ionic capacitor build android" from Visual Studio Code. When I open Android Studio and try to Sync Project with Gradle Files, I encounter the following error:

>FAILURE: Build failed with an exception.
>* What went wrong:
>A problem occurred configuring project ':capacitor-community-bluetooth-le'.
> Could not resolve all files for configuration ':capacitor-community-bluetooth-le:classpath'.
   > Could not download kotlin-gradle-plugin-model-1.8.20.jar (org.jetbrains.kotlin:kotlin-gradle-plugin-model:1.8.20)
      > Could not get resource 'https://repo.maven.apache.org/maven2/org/jetbrains/kotlin/kotlin-gradle-plugin-model/1.8.20/kotlin-gradle-plugin-model-1.8.20.jar'.
         > Could not GET 'https://repo.maven.apache.org/maven2/org/jetbrains/kotlin/kotlin-gradle-plugin-model/1.8.20/kotlin-gradle-plugin-model-1.8.20.jar'. Received status code 400 from server: Bad Request
   > Could not download kotlin-gradle-plugin-api-1.8.20-gradle76.jar (org.jetbrains.kotlin:kotlin-gradle-plugin-api:1.8.20)
      > Could not get resource 'https://repo.maven.apache.org/maven2/org/jetbrains/kotlin/kotlin-gradle-plugin-api/1.8.20/kotlin-gradle-plugin-api-1.8.20-gradle76.jar'.
         > Could not GET 'https://repo.maven.apache.org/maven2/org/jetbrains/kotlin/kotlin-gradle-plugin-api/1.8.20/kotlin-gradle-plugin-api-1.8.20-gradle76.jar'. Received status code 400 from server: Bad Request
   > Could not download kotlin-gradle-plugin-api-1.8.20.jar (org.jetbrains.kotlin:kotlin-gradle-plugin-api:1.8.20) [....]

To provide some context:

**The plugin version is 3**
"@capacitor-community/bluetooth-le": "^3.0.0"

**Java version is 17**
compileOptions {
sourceCompatibility JavaVersion.VERSION_17
targetCompatibility JavaVersion.VERSION_17
}

**My Gradle version is 8.0.2**
distributionUrl=https://services.gradle.org/distributions/gradle-8.0.2-all.zip

I have attempted to update the Gradle version, but it causes issues with other packages. The project was just created, and I'm not sure which steps to follow to get everything working correctly.

Thank you very much for the help :)

**Comment by pwespi** [comment]
I just created a new project and didn't have any problem. Maybe it was a temporary issue of the server at `repo.maven.apache.org`. I would just try again.

**Reply by mayte130** [reply]
Hi,
I have tried again in my project and I'm getting the same error.
To perform a test, I created a clean project and encountered the same error.
I am attaching the Android project in case you see any strange configuration that could be causing the issue.
Can you guide me in any direction?


>ionic start bluetooth blank --type=angular

I have installed the plugin and launched it in Android Studio
>npm install @capacitor-community/bluetooth-le
>ionic cap sync
>ionic capacitor build android

`Could not GET 'https://repo.maven.apache.org/maven2/org/jetbrains/kotlin/kotlin-gradle-plugin-model/1.8.20/kotlin-gradle-plugin-model-1.8.20.jar'. Received status code 400 from server: Bad Request`

Greetings,
Mayte
[android_Bluetooth.zip](https://github.com/capacitor-community/bluetooth-le/files/12318913/android_Bluetooth.zip)

![image](https://github.com/capacitor-community/bluetooth-le/assets/4377773/aa8b4ba3-49da-40c2-b432-bd211b45dbee)

**Reply by pwespi** [reply]
```sh
ionic start bluetooth blank --type=angular
cd bluetooth
npm install @capacitor-community/bluetooth-le
ionic cap sync
ionic capacitor build android
```

I tried your commands and didn't get any error.

Is your Android Studio up-to-date?

> Could not GET 'https://repo.maven.apache.org/maven2/org/jetbrains/kotlin/kotlin-gradle-plugin-model/1.8.20/kotlin-gradle-plugin-model-1.8.20.jar'. Received status code 400 from server: Bad Request

What happens if you open the link in the browser?

**Reply by mayte130** [reply]
I have updated the Android Studio version from Flamingo 2022.2.1 to Giraffe 2022.3.1, and I still have the same issue. It's very strange because when I try to open links in the browser, it does download the file for me.

![image](https://github.com/capacitor-community/bluetooth-le/assets/4377773/1c05813c-cbce-4c6d-ab86-cf7d7ce0437c)

I have tried everything, for example, opening Android Studio as an administrator, and it's still failing.
I don't know what else to do :(

**Comment by mayte130** [comment]
Hi @pwespi,
Sorry for the inconvenience, could you please check the Android Studio project configuration to see if you notice anything unusual that might be causing the error?
Thank you! 

![Captura de pantalla1](https://github.com/capacitor-community/bluetooth-le/assets/4377773/42464c42-2a2b-477b-8acc-268f88f2545a)
![Captura de pantalla2](https://github.com/capacitor-community/bluetooth-le/assets/4377773/c0c66375-4cba-4702-8d3b-903d3d98d07b)

**Comment by mayte130** [comment, marked as answer]
Hi, 
the problem was that I had the proxy configuration saved in an internal file. C:\Users\[username]\.gradle
Deleting or commenting these proxy configuration lines from the 'gradle.properties' file solves the problem.

#Tue Aug 08 12:55:14 CEST 2023
#systemProp.http.proxyHost=localhost
#systemProp.http.proxyPort=80
#systemProp.https.proxyHost=localhost
#systemProp.https.proxyPort=80

Thanks for your time :)
